service/create/service.go

```diff
@@ -794,10 +794,53 @@ func (s *Service) processCluster(cluster awstpr.CustomObject) error {
 		return microerror.Maskf(executionFailedError, fmt.Sprintf("could not make subnet public, '%#v'", err))
 	}
 
-	if err := publicSubnet.AddHostVPCRoute(routeTable, vpcPeeringConection.(*awsresources.VPCPeeringConnection)); err != nil {
+	conn, err := vpcPeeringConection.(*awsresources.VPCPeeringConnection).FindExisting()
+	if err != nil {
+		return microerror.Maskf(executionFailedError, fmt.Sprintf("could not find vpc peering connection: '%#v'", err))
+	}
+
+	routeTableID, err := routeTable.GetID()
+	if err != nil {
+		return microerror.Maskf(executionFailedError, fmt.Sprintf("could not get route table ID: '%#v'", err))
+	}
+
+	publicRoute := &awsresources.Route{
+		RouteTableID:         routeTableID,
+		DestinationCidrBlock: *conn.AccepterVpcInfo.CidrBlock,
+		VpcID:                *conn.VpcPeeringConnectionId,
+		AWSEntity:            awsresources.AWSEntity{Clients: clients},
+	}
+
+	if err := publicRoute.CreateOrFail(); err != nil {
 		return microerror.Maskf(executionFailedError, fmt.Sprintf("could not add host vpc route: '%#v'", err))
 	}
 
+	for _, privateSubnetSpec := range cluster.Spec.AWS.VPC.PrivateSubnets {
+
+		privateRouteTable := &awsresources.RouteTable{
+			Name:   privateSubnetSpec.RouteTableName,
+			VpcID:  cluster.Spec.AWS.VPC.PeerID,
+			Client: hostClients.EC2,
+		}
+
+		privateRouteTableID, err := privateRouteTable.GetID()
+		if err != nil {
+			return microerror.Maskf(executionFailedError, fmt.Sprintf("could not get private route table ID: '%#v'", err))
+		}
+
+		privateRoute := &awsresources.Route{
+			RouteTableID:         privateRouteTableID,
+			DestinationCidrBlock: *conn.RequesterVpcInfo.CidrBlock,
+			VpcID:                *conn.VpcPeeringConnectionId,
+			AWSEntity:            awsresources.AWSEntity{Clients: hostClients},
+		}
+
+		if err := privateRoute.CreateOrFail(); err != nil {
+			return microerror.Maskf(executionFailedError, fmt.Sprintf("could not add guest vpc route: '%#v'", err))
+		}
```

Comment:
I'm worried those `CreateOrFail` calls will cause errors after ResyncPeriod and operator pod restart. @rossf7 will know more.

Reply:
Yes creating the routes needs to be idempotent for when the cluster is reprocessed after the resync period.

For some other resources we tag them with the cluster ID and then look for a resource with that tag. That might help here so that `CreateIfNotExists` can be implemented.

Reply:
I've changed this and it now uses `CreateIfNotExists`, I tested this by deleting the `aws-operator` deployment and restarting it to see if it reprocesses a cluster without errors.